Assets/Scripts/OrchestraArmy/Level/Level.cs

```diff
@@ -0,0 +1,325 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class Level
+{
+    public enum gridSpace { empty, floor, wall, doorU, doorD, doorL, doorR };
+    public gridSpace[,] grid;
+    public int roomHeight, roomWidth;
+    public bool beaten;
+    public Vector2 roomSizeWorldUnits = new Vector2(150, 150);
+    public Vector2 offsetOfRoom = new Vector2(0, 0);
+    float worldUnitsInOneGridCell = 1;
+    struct walker
+    {
+        public Vector2 dir;
+        public Vector2 pos;
+    }
+    List<walker> walkers;
+    float chanceWalkerChangeDir = .7f, chanceWalkerSpawn = 0.2f;
+    float chanceWalkerDestoy = 0.2f;
+    int maxWalkers = 6;
+    private float percentToFill = 0.05f;
+    public Level()
+    {
+        Setup();
+        CreateFloors();
+        CreateWalls();
+        RemoveSingleWalls();
+        CreateDoors();
+    }
+    public Level(bool boss)
+    {
+        BossSettings();
+        Setup();
+        CreateFloors();
+        CreateWalls();
+        RemoveSingleWalls();
+        CreateDoors();
+    }
+
+    void BossSettings()
+    {
+        chanceWalkerChangeDir = .99f;
+        chanceWalkerSpawn = 0.8f;
+        chanceWalkerDestoy = 0.8f;
+        maxWalkers = 20;
+        percentToFill = 0.1f;
+    }
+    
+    void Setup()
+    {
+		this.beaten = false;
+        //find grid size
+        roomHeight = Mathf.RoundToInt(roomSizeWorldUnits.x / worldUnitsInOneGridCell);
+        roomWidth = Mathf.RoundToInt(roomSizeWorldUnits.y / worldUnitsInOneGridCell);
+        //create grid
+        grid = new gridSpace[roomWidth, roomHeight];
+        //set grid's default state
+        for (int x = 0; x < roomWidth - 1; x++)
+        {
+            for (int y = 0; y < roomHeight - 1; y++)
+            {
+                //make every cell "empty"
+                grid[x, y] = gridSpace.empty;
+            }
+        }
+        //set first walker
+        //init list
+        walkers = new List<walker>();
+        //create a walker 
+        walker newWalker = new walker();
+        newWalker.dir = RandomDirection();
+        //find center of grid
+        Vector2 spawnPos = new Vector2(Mathf.RoundToInt(roomWidth / 2.0f),
+                                        Mathf.RoundToInt(roomHeight / 2.0f));
+        newWalker.pos = spawnPos;
+        //add walker to list
+        walkers.Add(newWalker);
+    }
+    void CreateFloors()
+    {
+        int iterations = 0;//loop will not run forever
+        do
+        {
+            //create floor at position of every walker
+            foreach (walker myWalker in walkers)
+            {
+                grid[(int)myWalker.pos.x, (int)myWalker.pos.y] = gridSpace.floor;
+            }
+            //chance: destroy walker
+            int numberChecks = walkers.Count; //might modify count while in this loop
+            for (int i = 0; i < numberChecks; i++)
```

Comment:
Waarom zoveel for loops eigenlijk? zijn allemaal van 0 tot walkers.count zover ik kan zien, hoef je toch maar 1 keer door te lopen?